backend/de.metas.business/src/main/java/de/metas/project/workorder/undertest/WorkOrderProjectObjectUnderTestRepository.java

```diff
@@ -139,12 +162,36 @@ public List<WOProjectObjectUnderTest> getByQuery(@NonNull final WOObjectUnderTes
 				.collect(ImmutableList.toImmutableList());
 	}
 
+	@Nullable
+	public WOProjectObjectUnderTest getByOrderLineId(@NonNull final OrderLineId orderLineId)
+	{
+		return Optional.ofNullable(queryBL.createQueryBuilder(I_C_Project_WO_ObjectUnderTest.class)
+										   .addEqualsFilter(I_C_Project_WO_ObjectUnderTest.COLUMNNAME_C_OrderLine_Provision_ID, orderLineId.getRepoId())
+										   .create()
+										   .firstOptional(I_C_Project_WO_ObjectUnderTest.class))
+				.filter(Optional::isPresent)
+				.map(Optional::get)
+				.map(WorkOrderProjectObjectUnderTestRepository::ofRecord)
+				.orElse(null);
+	}
+
+	public void updateObjectDeliveredDate(@NonNull final WOProjectObjectUnderTest woProjectObjectUnderTest, @NonNull final I_C_OrderLine orderLine)
+	{
+		final I_C_Project_WO_ObjectUnderTest record = InterfaceWrapperHelper.loadByIds(ImmutableSet.of(woProjectObjectUnderTest.getObjectUnderTestId().getRepoId()), I_C_Project_WO_ObjectUnderTest.class).get(0);
+		record.setObjectDeliveredDate(orderLine.getDateDelivered());
+		InterfaceWrapperHelper.saveRecord(record);
+	}
```

Comment:
1. drop the method
2. set the ObjectDeliveredDate on `WOProjectObjectUnderTest` at the `Service` level
3. use the existing save/update method